doc/neps/nep-0056-array-api-main-namespace.rst

```diff
@@ -0,0 +1,524 @@
+other rules as expected, and work with all NumPy's dtypes rather than only the
+ones in the standard. The same goes for backwards-incompatible changes (e.g.,
+linear algebra functions need to all support batching in the same way, and
+consider the last two axes as matrices). As a result, NumPy should become more
+rather than less consistent.
+
+We'll note that one remaining incompatibility will be that NumPy is returning
+array scalars rather than 0-D arrays in most cases where the standard, and
+other array libraries, return 0-D arrays (e.g., indexing and reductions). There
+have been multiple discussions over the past year about the feasibility of
+removing array scalars from NumPy, or at least no longer returning them by
+default. However, this would be a large effort with some uncertainty about
+technical risks and impact of the change, and no one has taken it on. Given
+that array scalars implement a mostly array-compatible interface, this doesn't
+seem like the highest-prio item regarding array API standard compatibility.
+
+Here are what we see as the main expected benefits and costs of the complete
+set of proposed changes:
+
+Benefits:
+
+- It will remove the "having to make a choice between the NumPy API and the
+  Array API" issue for other libraries,
+- The array API standard tends to have more consistent behavior than NumPy
+  itself has (in cases where there are differences between the two, see for
+  example the `linear algebra design principles <https://data-apis.org/array-api/2022.12/extensions/linear_algebra_functions.html#design-principles>`__
+  and `data-dependent output shapes page <https://data-apis.org/array-api/2022.12/design_topics/data_dependent_output_shapes.html>`__
+  in the standard),
+- Easier for CuPy, JAX, PyTorch, Dask, Numba, and other such libraries and
+  compilers to fully match or support NumPy,
+- A few new features that have benefits independent of the standard: adding
+  ``matrix_transpose`` and ``ndarray.mT``, adding ``vecdot``, introducing
+  ``matrix_norm``/``vector_norm`` (they can be made gufuncs, vecdot already has
+  a PR making it one),
+
+Costs:
+
+- A number of backwards compatibility breaks (mostly minor, see the Backwards
+  compatibility section further down),
+- Expanding the size of the main namespace with about ~20 aliases (e.g.,
+  ``acos`` & co. with C99 names aliasing ``arccos`` & co.).
+
+Overall we believe that the benefits significantly outweigh the gains - and are
+permanent, while the costs are largely temporary. In particular, the benefits
+to array libraries and compilers that want to achieve compatibility with NumPy
+are significant. And as a result, the long-term benefits for the PyData (or
+scientific Python) ecosystem as a whole - because of downstream libraries being
+able to support multiple array libraries as easily as possible - are
+significant too. The number of breaking changes needed is fairly limited, and
```

Comment:
opinion: I'd use slightly weaker wording here, as reaching the "as easily as possible" point will require slightly more than just array API adoption.
```suggestion
are significant. And as a result, the long-term benefits for the PyData (or
scientific Python) ecosystem as a whole - because of downstream libraries being
able to support multiple array libraries much more easily - are
significant too. The number of breaking changes needed is fairly limited, and
```